cmd/mksyscall/main.go

```diff
@@ -62,7 +62,10 @@ func main() {
 // install makes sure mkwinsyscall can be called by
 // running go install golang.org/x/sys/windows/mkwinsyscall.
 func install(goTool string) {
-	args := []string{"install", "golang.org/x/sys/windows/mkwinsyscall"}
+	// mkwinsyscall is hardcoded here instead of adding it to go.mod so
+	// it doesn't appear in go.sum, which will reduce the likelihood
+	// of having patch conflicts when vendoring go-crypto-winnative.
+	args := []string{"install", "golang.org/x/sys/windows/mkwinsyscall@b874c991c1a5"}
```

Comment:
Thinking the `b874c991c1a5` should probably be a bit more visible... maybe moved to a var near the top of the file.

Also comes to mind that we could use `go-infra` as an indirection: `go install github.org/microsoft/go-infra/cmd/mkwinsyscallpassthrough@abcd123`, and use `mkwinsyscallpassthrough` as if it's `mkwinsyscall` so we can use the go.mod+go.sum over in go-infra for x/sys. I don't know if this really makes a difference, though, because at some point we're trusting `...@abcd123` either way. It seems to me it would just make it harder to update and end up being counterproductive. 😄 Figured I'd still mention it in case there's something I missed.

Reply:
> Thinking the b874c991c1a5 should probably be a bit more visible... maybe moved to a const near the top of the file.

It make sense 😸 

> Also comes to mind that we could use go-infra as an indirection: go install github.org/microsoft/go-infra/cmd/mkwinsyscallpassthrough@abcd123, and use mkwinsyscallpassthrough as if it's mkwinsyscall so we can use the go.mod+go.sum over in go-infra for x/sys. I don't know if this really makes a difference, though, because at some point we're trusting ...@abcd123 either way. It seems to me it would just make it harder to update and end up being counterproductive. 😄 Figured I'd still mention it in case there's something I missed.

If we do this, `go-infra` would appear in `go.sum` of this module, so it would also appear in our patch files, defeating the purpose of this PR.

Clarification: I'm not against adding dependencies in our crypto backend repos, but in this case having `x/sys` does not add much value, it reduces conflict chances, and not having it in the `go.mod`/`go.sum` clearly communicates to an outside reader that we don't depend on that at runtime.

Reply:
> If we do this, `go-infra` would appear in `go.sum` of this module, so it would also appear in our patch files, defeating the purpose of this PR.

I mean that `[]string{"install", "golang.org/x/sys/windows/mkwinsyscall@b874c991c1a5"}` would be replaced with `[]string{"install", "github.org/microsoft/go-infra/cmd/mkwinsyscallpassthrough@abcd123"}`, so still no go.mod dependency in this repo.

Reply:
oh, got it. I'll rather keep it simple and don't add the indirection 😄

Reply:
Yep, same--as far as I can tell, doesn't add any actual benefit (no go.sum in this repo, so it's not any safer, just a different endpoint being trusted). It would basically just make it take more steps to actually do an upgrade.